packages/components/text/src/Text.tsx

```diff
@@ -49,6 +50,8 @@ export const Text = compressible<TextProps, TextTokens>((props: TextProps, useTo
   // get the tokens from the theme
   let [tokens, cache] = useTokens(theme);
 
+  const fontMetricsScaleFactors = Platform.OS === 'ios' ? useFontMetricsScaleFactors() : undefined;
```

Comment:
hooks must run on every render, so you need to call useFontMetricsScaleFactors unconditionally

Reply:
Right, thanks for catching. A `Platform` check is effectively unconditional but eslint doesn't recognize it as such. How we've solved this elsewhere is to move `if` checks to inside the custom hook: https://github.com/microsoft/fluentui-react-native/blob/14465a397887dc9cbe29bedd35461a3356be2cbf/packages/components/Notification/src/Notification.tsx#L24

Also worth linking again: https://reactjs.org/docs/hooks-rules.html

Reply:
Resolved with latest changes.